ios/FluentUI/Core/Colors2.swift

```diff
@@ -0,0 +1,243 @@
+        brandColors[.brandForeground1Pressed] = brandForeground1Pressed
+    }
+    if let brandForeground1Selected = provider.brandForeground1Selected(for: theme)?.dynamicColor {
+        brandColors[.brandForeground1Selected] = brandForeground1Selected
+    }
+    if let brandForegroundTint = provider.brandForegroundTint(for: theme)?.dynamicColor {
+        brandColors[.brandForegroundTint] = brandForegroundTint
+    }
+    if let brandForegroundDisabled1 = provider.brandForegroundDisabled1(for: theme)?.dynamicColor {
+        brandColors[.brandForegroundDisabled1] = brandForegroundDisabled1
+    }
+    if let brandForegroundDisabled2 = provider.brandForegroundDisabled2(for: theme)?.dynamicColor {
+        brandColors[.brandForegroundDisabled2] = brandForegroundDisabled2
+    }
+    if let brandStroke1 = provider.brandStroke1(for: theme)?.dynamicColor {
+        brandColors[.brandStroke1] = brandStroke1
+    }
+    if let brandStroke1Pressed = provider.brandStroke1Pressed(for: theme)?.dynamicColor {
+        brandColors[.brandStroke1Pressed] = brandStroke1Pressed
+    }
+    if let brandStroke1Selected = provider.brandStroke1Selected(for: theme)?.dynamicColor {
+        brandColors[.brandStroke1Selected] = brandStroke1Selected
+    }
+    return brandColors
+}
+
+// MARK: Colors
+
+@objc(MSFColors2)
+public final class Colors2: NSObject {
+    /// Associates a `ColorProvider2` with a given `UIWindow` instance.
+    ///
+    /// - Parameters:
+    ///   - provider: The `ColorProvider2` whose colors should be used for controls in this window.
+    ///   - window: The window where these colors should be applied.
+    @objc public static func setProvider(provider: ColorProviding2, for theme: FluentTheme) {
+        colorProvidersMap.setObject(provider, forKey: theme)
+
+        // Create an updated fluent theme as well
+        let brandColors = brandColorOverrides(provider: provider, for: theme)
+        brandColors.forEach { token, value in
+            theme.aliasTokens.colors[token] = value
+        }
+    }
+
+    /// Removes any associated `ColorProvider2` from the given `UIWindow` instance.
+    ///
+    /// - Parameters:
+    ///   - window: The window that should have its `ColorProvider2` removed.
+    @objc public static func removeProvider(for theme: FluentTheme) {
+        colorProvidersMap.removeObject(forKey: theme)
+        // TODO: what is the equivalent function here?
+        // window.fluentTheme = FluentThemeKey.defaultValue
+    }
+
+    // MARK: Primary
+
+    /// Use these funcs to grab a color customized by a ColorProvidingobject for a specific window. If no colorProvider2 exists for the window, falls back to deprecated singleton theme color
+    @objc public static func brandBackground1(for theme: FluentTheme) -> UIColor {
+        return colorProvidersMap.object(forKey: theme)?.brandBackground1(for: theme) ?? FallbackThemeColor.brandBackground1
```

Comment:
Proposal:
* Remove `colorProvidersMap` variable entirely (this is a holdover from the version of `Colors.swift` that predates `FluentTheme`).
* Remove all of these static methods, along with `FallbackThemeColor`, entirely. Now that `AliasTokens` are accessible via Objective-C, clients can use that approach to fetch values.

Before:
```objc
MSFFluentTheme *fluentTheme = [[self view] fluentTheme];
UIColor *brandBackgroundColor = [MSFColors2 brandBackground1For:fluentTheme];
```

After:
```objc
MSFFluentTheme *fluentTheme = [[self view] fluentTheme];
UIColor *brandBackgroundColor = [[fluentTheme aliasTokens] colorsForToken:MSFColorsAliasTokensBackground1];

Reply:
Are you saying we don't need the getters any more, just the setters? I agree with you, if that's the case, and is actually how I'm doing it in the demo controller test code.

Reply:
Done!